api/envoy/extensions/wasm/v3/wasm.proto

```diff
@@ -131,6 +132,12 @@ message VmConfig {
   // on native platforms.
   // Warning: Envoy rejects the configuration if there's conflict of key space.
   EnvironmentVariables environment_variables = 7;
+
+  // Secret that contains image pull secret for fetching Wasm plugin from an OCI registry.
+  //
+  // [#comment:TODO(jewertow): Revert hiding once all major OCI registries are supported.]
```

Comment:
From api's perspective, should we add the enum to indicate which registry is used, like Docker, Azure, and so on?
It seems not all registry need the secret, and we can use config to distinguish.

Reply:
All private registries, regardless of the provider, will need the image pull secret. In case of public registries we don't need the secret. So currently, the client does not perform authorization if the secret is not specified.

I think we are able to determine the provider in the most cases, so I wanted to avoid extending the API if that's not really necessary.